### PR TITLE
ENH: metrics in secret extension

### PR DESCRIPTION
**Pull request by chenqi0805**
### Description
This PR adds metrics on AWS secrets extension plugin
 
### Issues Resolved
Resolves #[Issue number to be closed when this PR is merged]
 
### Check List
- [ ] New functionality includes testing.
- [ ] New functionality has a documentation issue. Please link to it in this PR.
  - [ ] New functionality has javadoc added
- [ ] Commits are signed with a real name per the DCO

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/data-prepper/blob/main/CONTRIBUTING.md).

### CODE REVIEW

File: data-prepper-plugins/aws-plugin/src/main/java/org/opensearch/dataprepper/plugins/aws/AwsSecretPlugin.java [hunk 1]
```diff
@@ -33,6 +35,7 @@ public AwsSecretPlugin(final AwsSecretPluginConfig awsSecretPluginConfig) {
             this.pluginConfigPublisher = new AwsSecretsPluginConfigPublisher();
             pluginConfigValueTranslator = new AwsSecretsPluginConfigValueTranslator(secretsSupplier);
             scheduledExecutorService = Executors.newSingleThreadScheduledExecutor();
+            pluginMetrics = PluginMetrics.fromNames("secrets", "aws");
```
Comment (dlvenable) [inline data-prepper-plugins/aws-plugin/src/main/java/org/opensearch/dataprepper/plugins/aws/AwsSecretPlugin.java:38]: I tend to think that this should be `fromNames("aws", "secrets")`

I'm not sure that we will even have other secrets, so this seems to create an unnecessary extra layer.

**Review by dlvenable — CHANGES_REQUESTED** [review]

File: data-prepper-plugins/aws-plugin/src/main/java/org/opensearch/dataprepper/plugins/aws/AwsSecretPlugin.java (hunk 1 above)
Comment (chenqi0805) [reply data-prepper-plugins/aws-plugin/src/main/java/org/opensearch/dataprepper/plugins/aws/AwsSecretPlugin.java:38]: This is to be consistent with https://github.com/opensearch-project/data-prepper/blob/e6df3eb2cd46ebd13dd1c7b808d288c2f3d6ee51/data-prepper-core/src/main/java/org/opensearch/dataprepper/peerforwarder/certificate/CertificateProviderFactory.java#L68

File: data-prepper-plugins/aws-plugin/src/main/java/org/opensearch/dataprepper/plugins/aws/SecretsRefreshJob.java [hunk 2]
```diff
@@ -1,23 +1,53 @@
...
 
     @Override
     public void run() {
-        secretsSupplier.refresh(secretConfigId);
-        pluginConfigPublisher.notifyAllPluginConfigObservable();
+        secretsRefreshTimer.record(() -> {
+            try {
+                secretsSupplier.refresh(secretConfigId);
+                secretsRefreshSuccessCounter.increment();
+                pluginConfigPublisher.notifyAllPluginConfigObservable();
```
[the remark this answers is not in the document]
Comment (chenqi0805) [reply data-prepper-plugins/aws-plugin/src/main/java/org/opensearch/dataprepper/plugins/aws/SecretsRefreshJob.java:46]: Yes. I include it in the timer to track the total time. But this operation should all be in-memory and I expect it to be much less time consuming compared to get secret call

File: data-prepper-plugins/aws-plugin/src/main/java/org/opensearch/dataprepper/plugins/aws/AwsSecretPlugin.java (hunk 1 above)
Comment (dlvenable) [reply data-prepper-plugins/aws-plugin/src/main/java/org/opensearch/dataprepper/plugins/aws/AwsSecretPlugin.java:38]: The documentation indicates that we go from a componentId to a componentScope.

https://github.com/opensearch-project/data-prepper/blob/1af1ce9c6684994aaac059a4f79b8a780f8a09a8/data-prepper-api/src/main/java/org/opensearch/dataprepper/metrics/PluginMetrics.java#L36-L37

I think the component in this case is the aws-plugin. So I would be in favor of using `aws` as the componentId. 

@asifsmohammed , Any thoughts from you end?

Comment (asifsmohammed) [reply data-prepper-plugins/aws-plugin/src/main/java/org/opensearch/dataprepper/plugins/aws/AwsSecretPlugin.java:38]: I think what George has in the PR is right order. We do `<componentScope> + "." + <componentId>`

**Review by dlvenable — APPROVED** [review]

**Review by oeyh — APPROVED** [review]

**Pull request merged**